src/generator/AutoRest.Ruby.Azure/CodeGeneratorRba.cs

```diff
@@ -73,8 +72,11 @@ public CodeGeneratorRba()
                     continue;
                 }
 
-                var modelTemplate = new ModelTemplate { Model = model };
-                await Write(modelTemplate, Path.Combine(GeneratorSettingsRb.Instance.modelsPath, CodeNamer.UnderscoreCase(model.Name) + ImplementationFileExtension));
+                var modelTemplate = new AzureModelTemplate { Model = model };
```

Comment:
Looking into AzureModelTemplate the only difference I saw with Model template is an if statement asking if an accessor should be generated and avoid generating it if the model is Resource or Subsresource, is this correct? If so, could we be making the distinction here and producing the template with the accessor (AzureModelTemplate) if the model matches a resource or subsresource, and produce a Model template in the rest of the cases?
or is there another way where we could avoid repeating the code between ModelTemplate and AzureModelTemplate and add a condition to ModelTemplate that can produce the difference?